src/types/NamingConventionConfig.ts

```diff
@@ -1,34 +1,66 @@
 import { DiagnosticConfig } from './StylisticConfig'
 
-export type NamingConvention = 'camelCase' | 'PascalCase' | 'snake_case' | 'SCREAMING_SNAKE_CASE' | 'kebab-case'
-export type NamingConventionConfig = NamingConvention | NamingConvention[]
+export const DefaultNamingConventions = ['camelCase', 'PascalCase', 'snake_case', 'SCREAMING_SNAKE_CASE', 'kebab-case', 'lowercase', 'UPPERCASE'] as const
+export type DefaultNamingConvention = typeof DefaultNamingConventions[number]
+export type NamingConventionConfig = string | ParentCustomNamingConvention | (string | ParentCustomNamingConvention)[]
+export interface ParentCustomNamingConvention extends CustomNamingConvention {
+    label: string
+}
+
+interface CustomNamingConvention {
+    prefix?: string,
+    suffix?: string,
+    allowLessParts?: boolean,
+    allowMoreParts?: boolean,
+    parts: string | CustomNamingConvention | string[] | CustomNamingConvention[],
+    sep: string
+}
 
-const validators: Record<NamingConvention, RegExp> = {
+const defaultValidators: Record<DefaultNamingConvention, RegExp> = {
     camelCase: /^[a-z\d]+([A-Z\d]+[a-z\d]*)*$/,
     PascalCase: /^([A-Z\d]+[a-z\d]*)+$/,
     snake_case: /^[a-z\d]+(_[a-z\d]+)*$/,
     SCREAMING_SNAKE_CASE: /^[A-Z\d]+(_[A-Z\d]+)*$/,
-    'kebab-case': /^[a-z\d]+(\-[a-z\d]+)*$/
+    'kebab-case': /^[a-z\d]+(\-[a-z\d]+)*$/,
+    lowercase: /^[a-z\d]+$/,
+    UPPERCASE: /^[A-Z\d]+$/
 }
 
 /**
  * Check whether the identity follows the naming convention or not.
  * @param identity An identity.
  * @param config A naming convention config.
  */
-export function checkNamingConvention(identity: string, config: DiagnosticConfig<NamingConventionConfig>) {
-    if (!config) {
-        return true
-    }
-    let value = config[1]
-    if (!(value instanceof Array)) {
-        value = [value]
-    }
-    for (const convention of value) {
-        const validator = validators[convention]
-        if (validator.test(identity)) {
+export function checkNamingConvention(identity: string, config: DiagnosticConfig<NamingConventionConfig>): boolean {
+    if (!config) return true
```

Comment:
Actually let's change this (and all other `if` statements here) back to using curly braces (`{}`).